code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -844,6 +845,359 @@
+	/// Voting for King
+	var/list/mob/living/carbon/xenomorph/votes = list()
+
+/obj/effect/alien/resin/king_cocoon/Destroy()
+	if(!hatched)
+		marine_announcement("ALERT.\n\nUNSUAL ENERGY BUILDUP IN [uppertext(get_area_name(loc))] HAS BEEN STOPPED.", "[MAIN_AI_SYSTEM] Biological Scanner", 'sound/misc/notice1.ogg')
+		var/datum/hive_status/hive
+		for(var/cur_hive_num in GLOB.hive_datum)
+			hive = GLOB.hive_datum[cur_hive_num]
+			if(!length(hive.totalXenos))
+				continue
+			if(cur_hive_num == hive_number)
+				xeno_announcement(SPAN_XENOANNOUNCE("THE HATCHERY WAS DESTROYED! VENGENCE!"), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+			else
+				xeno_announcement(SPAN_XENOANNOUNCE("THE HATCHERY WAS DESTROYED!"), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+
+	var/datum/hive_status/hive = GLOB.hive_datum[hive_number]
+	hive.has_hatchery = FALSE
+	for(var/obj/effect/alien/resin/special/pylon/pylon as anything in hive.active_endgame_pylons)
+		pylon.protection_level = initial(pylon.protection_level)
+		pylon.update_icon()
+
+	votes = null
+	chosen_candidate = null
+
+	. = ..()
+
+/obj/effect/alien/resin/king_cocoon/Initialize(mapload, hivenumber)
+	. = ..()
+	if(hivenumber)
+		hive_number = hivenumber
+	
+	var/datum/hive_status/hatchery_hive = GLOB.hive_datum[hive_number]
+	hatchery_hive.has_hatchery = TRUE
+	color = hatchery_hive.color
+
+	for(var/x_offset in -1 to 1)
+		for(var/y_offset in -1 to 1)
+			var/turf/turf_to_block = locate(x + x_offset, y + y_offset, z)
+			var/obj/effect/build_blocker/blocker = new(turf_to_block, src)
+			blockers += blocker
+
+	timer = addtimer(CALLBACK(src, PROC_REF(start_growing)), 10 SECONDS, TIMER_UNIQUE|TIMER_STOPPABLE|TIMER_DELETE_ME)
+	addtimer(CALLBACK(src, PROC_REF(check_pylons)), 10 SECONDS, TIMER_UNIQUE|TIMER_STOPPABLE|TIMER_DELETE_ME|TIMER_LOOP)
+
+	marine_announcement("ALERT.\n\nUNSUAL ENERGY BUILDUP DETECTED IN [uppertext(get_area_name(loc))].\n\nESTIMATED TIME UNTIL COMPLETION - 10 MINUTES.", "[MAIN_AI_SYSTEM] Biological Scanner", 'sound/misc/notice1.ogg')
+	var/datum/hive_status/hive
+	for(var/cur_hive_num in GLOB.hive_datum)
+		hive = GLOB.hive_datum[cur_hive_num]
+		if(!length(hive.totalXenos))
+			continue
+		if(cur_hive_num == hive_number)
+			xeno_announcement(SPAN_XENOANNOUNCE("The King is growing at [get_area_name(loc)]. Protect it at all costs!"), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+		else
+			xeno_announcement(SPAN_XENOANNOUNCE("Another hive's King is growing at [get_area_name(loc)]."), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+
+/// Callback for a repeating 10s timer to ensure both pylons are active (otherwise delete) and counts the number of marines groundside (would cause hatching to expedite).
+/obj/effect/alien/resin/king_cocoon/proc/check_pylons()
+	var/datum/hive_status/hive = GLOB.hive_datum[hive_number]
+
+	if(length(hive.active_endgame_pylons) < 2)
+		qdel(src)
+		return
+
+	if(chosen_candidate || rolling_candidates)
+		return
+
+	if(icon_state == "hatching" || icon_state == "hatched")
```

Comment:
"hatching" is only ever set by flick(), which doesn't cause an actual change to the vars (https://www.byond.com/docs/ref/#/proc/flick), and icon_state == "hatched" seems it could be better checked by just checking `hatched`?

```suggestion
	if(hatched)
```